### PR TITLE
Typescript+Axios: multipart/form-data correctly handled

### PR DESCRIPTION
**Pull request by mvniekerk**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10)

### Description of the PR
When doing a multipart/formdata request, the current state is that it changes the content type to ''application/x-www-form-urlencoded" (violating the contract) and then does a toString() on the request data.

What this PR does is to use FormData as the data, it appends (instead of doing a .set()) and then omits the toString() when _the content-type is multipart/form-data_

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-axios/package.mustache [hunk 1]
```diff
@@ -24,7 +24,7 @@
   },
   "devDependencies": {
     "@types/node": "^8.0.9",
-    "typescript": "^2.4" 
+    "typescript": "^3.2"
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-axios/package.mustache]: which feature requires typescript 3.2? I would widen the restriction as much as possible

File: samples/client/petstore/typescript-axios/builds/default/api.ts [hunk 2]
```diff
@@ -473,27 +476,28 @@ export const PetApiAxiosParamCreator = function (configuration?: Configuration)
...
-					? configuration.accessToken("petstore_auth", ["write:pets", "read:pets"])
-					: configuration.accessToken;
+                const localVarAccessTokenValue = typeof configuration.accessToken === 'function'
+                    ? configuration.accessToken("petstore_auth", ["write:pets", "read:pets"])
+                    : configuration.accessToken;
                 localVarHeaderParameter["Authorization"] = "Bearer " + localVarAccessTokenValue;
             }
 
             if (tags) {
                 localVarQueryParameter['tags'] = tags.join(COLLECTION_FORMATS["csv"]);
```
Comment (macjohnny) [inline samples/client/petstore/typescript-axios/builds/default/api.ts]: should be adapted too, for consistency

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java [hunk 3]
```diff
@@ -126,6 +126,18 @@ public String getTypeDeclaration(Schema p) {
         }
     }
 
+    @Override
+    public Map<String, Object> postProcessOperations(Map<String, Object> objs) {
+        objs = super.postProcessOperations(objs);
+        Map<String, Object> vals = (Map<String, Object>)objs.get("operations");
+        List<CodegenOperation> operations = (List<CodegenOperation>)vals.get("operation");
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java]: add comment what this does

File: modules/openapi-generator/src/main/resources/typescript-axios/package.mustache (hunk 1 above)
Comment (mvniekerk) [reply modules/openapi-generator/src/main/resources/typescript-axios/package.mustache]: Reverted at 9818ee80b249d388b61211f5fccddf8dad2c8eea -
Thought the spread operators was a later language feature

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java (hunk 3 above)
Comment (mvniekerk) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java]: Addressed in 3ba25dc7ebc6a5c477a902fdbf3c7dc9ef7cbe69

File: samples/client/petstore/typescript-axios/builds/default/api.ts (hunk 2 above)
Comment (mvniekerk) [reply samples/client/petstore/typescript-axios/builds/default/api.ts]: Addressed in 3ba25dc

**Comment by wing328** [comment]
@mvniekerk the build failed with the following errors:
```
Status: Downloaded newer image for maven:3-jdk-8
[ERROR] COMPILATION ERROR : 
[ERROR] /gen/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:[129,5] method does not override or implement a method from a supertype
[ERROR] /gen/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:[131,21] cannot find symbol
  symbol: method postProcessOperations(java.util.Map<java.lang.String,java.lang.Object>)
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project openapi-generator: Compilation failure: Compilation failure: 
[ERROR] /gen/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:[129,5] method does not override or implement a method from a supertype
[ERROR] /gen/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:[131,21] cannot find symbol
[ERROR]   symbol: method postProcessOperations(java.util.Map<java.lang.String,java.lang.Object>)
[ERROR] -> [Help 1]
```

Please take a look and let us know if you need help fixing the issue.

**Comment by wing328** [comment]
Update: Pushed https://github.com/OpenAPITools/openapi-generator/pull/2002/commits/061cea51a58b5a0a1060d02cb7ccb1693e79870e to fix the compilation error. Let's see how it goes.

**Comment by wing328** [comment]
All CI tests passed 👍

**Pull request merged**